common/src/metrics.c

```diff
@@ -17,9 +17,30 @@
 #include "metrics.h"
 
 #include <stdint.h>
+#include <generated/csr.h>
 
 void get_csr_metrics(uint32_t *acc, uint32_t *refill, uint32_t *stall) {
   __asm__ volatile ("csrr %0, %1" : "=r"(*acc) : "i"(CSR_ACC_COUNTER));
   __asm__ volatile ("csrr %0, %1" : "=r"(*refill) : "i"(CSR_REFILL_COUNTER));
   __asm__ volatile ("csrr %0, %1" : "=r"(*stall) : "i"(CSR_STALL_COUNTER));
 }
+
+void set_flash_control(uint32_t val) {
+#ifdef LITESPI_CS_COUNTER
+  counter_control_write(val);
+#endif
+}
+
+void get_flash_ticks(void) {
+#ifdef LITESPI_CS_COUNTER
+  uint32_t* ticks_addr = (uint32_t*)(CSR_COUNTER_COUNTER_ADDR);
+#endif
+
+#ifdef LITESPI_CS_COUNTER
+  printf("[CS active] %llu out of %llu cycles\n",
```

Comment:
Hmm if I'm understanding the meaning of the counters correctly, I think this wording is a bit unclear. We are counting CS active in terms of system clock cycles (which is good, because then we can compare it to the total cycle count for the model evaluation and get a proportion of time it was active for). But the second number is SPI clock cycles not system clock cycles. Not really the same type of cycles.

Reply:
Right, we've changed it so that we're counting the CS# line in terms of system clock cycles *and* we're now counting system clock cycles as well, regardless whether CS# line is asserted or not.